napari/layers/image/image.py

```diff
@@ -527,18 +527,46 @@ def interpolation(self):
         str
             The current interpolation mode
         """
-        return str(self._interpolation[self._ndisplay])
+        return str(
+            self._interpolation2d
+            if self._ndisplay == 2
+            else self._interpolation3d
+        )
 
     @interpolation.setter
     def interpolation(self, interpolation):
         """Set current interpolation mode."""
+        warnings.warn(
+            trans._(
+                "Interpolation setting is deprecated. Please use interpolation2d or interpolation3d",
+            ),
+            category=FutureWarning,
```

Comment:
Same comment, if you say "deprecated", I guess the category should be  `DeprecationWarning` and the message should say since when.

```suggestion
            category=DeprecationWarning,
```

Also thanks for stacklevel !

Reply:
The problem with deprecation Warning is that they are not visible during the normal run of python code. So if the plugin creator does not test properly his code against deprecation he will do not know that he use deprecated code https://docs.python.org/3/library/exceptions.html#DeprecationWarning

Reply:
Hum I see your point, but I think if we want to make those warnings visible in napari, we  need to insert a `DeprecationWarning` filter to default at napari startup, potentially with a setting option to disable that. We can limit it to `module='napari.*'` to only target our modules.

Reply:
I see your point, but current strategy in napari codebase is to use FutureWarning.

https://github.com/napari/napari/blob/9cfb915e01a69e9717fad193533d0a6bcb92d98f/napari/qt/progress.py#L5

https://github.com/napari/napari/blob/d490354431bf265d580e73f15b7bfebdfb45e48e/napari/_qt/qt_event_loop.py#L265

I like the idea of your change, but maybe it is worth opening a separate PR/issue for this to allow discussion in the wider group?

Reply:
opened, https://github.com/napari/napari/issues/4731

> I see your point, but current strategy in napari codebase is to use FutureWarning.

But at the same time there is half a dozen DeprecationWarnings already emitted in napari/layers/utils/text_manager.py and a few others in napari/layers/base/base.py, napari/layers/utils/layer_utils.py , napari/resources/__init__.py


I think in total there are more Deprecation Warnings than FutureWarning. In any case I agree that we should be consistent.

Reply:
Done

Reply:
I was even aware of `FutureWarning`, so thanks for the info, and thanks for #4731 too!